### PR TITLE
Add new filter for meta field order

### PR DESCRIPTION
**Pull request by jacobarriola**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Adds a new filter so that consumer can handle types before the SQL query is set. There are instances where a float is being passed but the type value is set to null. This creates unexpected SQL results.

An example implementation would be something like this:

```php
/**
 * Ensure meta field type is set to decimal (float)
 */
add_filter( 'wpgraphql_cursor_ordering_field', function ( $fields, $builder, $object_cursor ) {
	// Bail if not a float
	if ( empty( strpos( $fields['value'], '.' ) ) ) {
		return $fields;
	}
	
	
	$fields['type'] = 'DECIMAL(10,' . wc_get_price_decimals() . ')';
	
	return $fields;
}, 10, 3 );
```

Does this close any currently open issues?
------------------------------------------
#1545 

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
See issue.

### CODE REVIEW

**Comment by jacobarriola** [comment]
@jasonbahl Do you need me to address some of these failing tests? Feels like issues outside the scope of this PR. LMK.

File: src/Data/Cursor/CursorBuilder.php [hunk 1]
```diff
@@ -21,28 +21,34 @@ public function __construct( $compare = '>' ) {
...
 		 * escape them here.
 		 */
-		$this->fields[] = [
-			'key'   => esc_sql( $key ),
-			'value' => esc_sql( $value ),
-			'type'  => esc_sql( $type ),
-			'order' => esc_sql( $order ),
-		];
+		$this->fields[] = apply_filters(
+			'wpgraphql_cursor_ordering_field',
```
Comment (jasonbahl) [inline src/Data/Cursor/CursorBuilder.php:42]: @jacobarriola can we get a docblock over the cursor explaining the params?

something like: 

```
/**
 * Filters the field used for ordering when cursors are used for pagination
 * 
 * @param array $field The field key, value, type and order 
 * @param CursorBuilder $this The CursorBuilder class
 * @param mixed null | PostObjectCursor $cursor The PostObjectCursor class
 */

**Review by jasonbahl — CHANGES_REQUESTED** [review]
see: https://github.com/wp-graphql/wp-graphql/pull/1546/files#r518813086

**Comment by jasonbahl** [comment]
@jacobarriola it looks like the tests are failing because of some composer dependency that needs updated. I'm working on getting that fixed now.

**Comment by jasonbahl** [comment]
@jacobarriola I fixed the composer issue that was messing up tests and merged it to your branch so tests are re-running now.

File: src/Data/Cursor/CursorBuilder.php (hunk 1 above)
Comment (jasonbahl) [reply src/Data/Cursor/CursorBuilder.php:42]: @jacobarriola upon looking at this a bit more, we also need to make sure we're escaping _after_ the filter. 

So, I think we need to change this a bit more too. 

Something like: 

```php
$field = [
  'key'   => esc_sql( $key ),
  'value' => esc_sql( $value ),
  'type'  => esc_sql( $type ),
  'order' => esc_sql( $order ),
];

$filtered_field = apply_filters( 'wpgraphql_cursor_ordering_field', $field, $this, $object_cursor );
$clean_fields = [];
if ( ! empty( $field ) && is_array( $field ) ) {
  foreach( $field as $key => $value ) {
     $clean_fields[ $key ] = esc_sql( $value );
  }
}

$this->fields = $clean_fields;

```

Comment (jasonbahl) [reply src/Data/Cursor/CursorBuilder.php:42]: ^ untested, just wrote here in Github as pseudo code, but might be legit 🤷‍♂️

**Review by jasonbahl — DISMISSED** [review]

File: src/Data/Cursor/CursorBuilder.php [hunk 2]
```diff
@@ -21,28 +21,57 @@ public function __construct( $compare = '>' ) {
...
+		 * @param null | PostObjectCursor $object_cursor The PostObjectCursor class
 		 */
-		$this->fields[] = [
-			'key'   => esc_sql( $key ),
-			'value' => esc_sql( $value ),
-			'type'  => esc_sql( $type ),
-			'order' => esc_sql( $order ),
-		];
+		$field = apply_filters(
+			'wpgraphql_cursor_ordering_field',
```
Comment (jasonbahl) [inline src/Data/Cursor/CursorBuilder.php:46]:
```suggestion
			'graphql_cursor_ordering_field',
```

**Pull request merged**